src/main/java/com/puppycrawl/tools/checkstyle/meta/MetadataGeneratorUtil.java

```diff
@@ -22,21 +22,36 @@
 import java.io.File;
 import java.io.IOException;
 import java.io.OutputStream;
+import java.lang.reflect.Field;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import org.apache.maven.doxia.macro.MacroExecutionException;
```

Comment:
It isn't ideal that the metadata module uses the site module. I think it's better if the opposite is the case - the site module uses the metadata module. Shall we deal with this in a follow-up issue?

Reply:
We will refactor later on to move methods around to make it better design wise.
For now let's focus on functionality to unblock ability to remove properties from javadoc

Reply:
Let's collect ideas at https://github.com/checkstyle/checkstyle/issues/13699 to improve in separate PRs